students/HABTAMU/lesson09/assignment/charges_calc.py

```diff
@@ -0,0 +1,134 @@
+
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+
+import logging
+import traceback
+import functools
+import sys
+
+def create_logger():
+    """
+    Creates a logging object and returns it, parse logging argument selective, by using decorators.
+    conditional logging with single command line variable which can turn logging on or off for decorated classes or functions.
+    """
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    formatter = logging.Formatter(log_format)
+    
+    # create the logging file handler, log filename format with timestamp 
+    file_handler = logging.FileHandler(datetime.datetime.now().strftime(" charges_calc %Y-%m-%d") + '.log')
+    file_handler.setFormatter(formatter)
+
+    # add handler to logger object
+    logger.addHandler(file_handler)
+
+    return logger
+
+
+def exception(function):
+    """
+    A decorator that wraps the passed in function and logs 
+    exceptions should one occur
+    """
+    @functools.wraps(function)
+    def wrapper(*args, **kwargs):
+        logger = create_logger()
+        try:
+            return function(*args, **kwargs)
+        except:
+            # log the exception
+            err = "There was an exception in  "
+            err += function.__name__
+            logger.exception(err)
+
+            # re-raise the exception
+            raise
+    return wrapper
```

Comment:
While it is true that the reserved word is Exception, with a capital E, I would still avoid using exception as a function name to avoid confusion.

Fyi, I would never do this as an actual decorator. Exceptions should always be logged, not toggled by a decorator. I think the assignment wanted to toggle all logging for those functions, not just exceptions, but the directions were not terribly clear.